### ISSUE TITLE
[Bug] unnecessary-string-conversion should not be raised in if condition

### ISSUE THREAD

**Issue by Leemur89** [MarketSquare/robotframework-robocop#952]
### What happened?

I am using a ``Set variable if`` with a condition based on the value of an environment variable. Without quotes around the variable python won't recognize it as a string but as a variable name

### What command/code did you try to run?

```robotframework
    ${output}       Set Variable If
    ...             any(substring in "%{ENV_VALUE}" for substring in @{PROTECTED_ENV_VALUES})
    ...             ${TRUE}
    ...             ${FALSE}
```

### What is the full error message?

unnecessary-string-conversion is raised

### What did you expect to happen instead?

unnecessary-string-conversion should not be raised

### Robocop version

4.1.0

**Comment by bhirsz** [comment]
What about using ``%ENV_VALUE``? (without brackets) Or only $.. supports passing variable directly?

**Comment by Leemur89** [comment]
I've tried that and unfortunately it does not get evaluated, only $ supports the removal of brackets in python evaluation

**Comment by bhirsz** [comment]
Thanks for confirming this. I will also check this and adjust the rule so it doesn't raise if it's impossible to avoid string conversion. And environment variables are strings already so there is no "unnecessary string conversion" anyway :)

**Comment by bhirsz** [comment]
Released in 4.2.0
